### PR TITLE
fix: (sfui2-1147) add animated drawer example from the right

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue

https://vsf.atlassian.net/browse/SFUI2-1147?atlOrigin=eyJpIjoiNDcxM2ExNjBjYzZhNDRlMGJmMWRhZjk0MzcxODAxNzAiLCJwIjoiaiJ9

# Scope of work

- added animated examples with radio controls to select placement
- `noScale` prop was added for Showcase
- on react there is a "sliding" effect when changing the radio value - to remove it it would require from us to add `isTransitioning` state in the example, but it may mislead the users on the variable purpose, therefore we decided to leave it as it s for now (@dkacper)

# Screenshots of visual changes

-

# Checklist

- [x] Self code-reviewed
- [x] Changes documented
- [x] Semantic HTML
- [x] SSR-friendly
- [x] Caching friendly
- [x] a11y for WCAG 2.0 AA
- [x] examples created
- [x] blocks created
- [x] cypress tests created

### CODE REVIEW

File: apps/preview/next/pages/showcases/Drawer/TransitionAndCloseButton.tsx [hunk 1]
```diff
@@ -1,70 +1,105 @@
...
+          <SfRadio
+            name="placement"
+            value="right"
+            checked={checkedPlacement === 'right'}
+            onChange={(event) => {
+              setCheckedPlacement(event.target.value as SfDrawerProps['placement']);
+            }}
+          />
+          <span className="ml-2">Slide from right</span>
+        </label>
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/Drawer/TransitionAndCloseButton.tsx:49]: I think we could make it an iterable options.

```suggestion
      {options.map(({ label, value }) => (
        <label className="flex items-center my-4 cursor-pointer">
          <SfRadio
            name="placement"
            value={value}
            checked={checkedPlacement === value}
            onChange={(event) => {
              setCheckedPlacement(event.target.value as SfDrawerProps['placement']);
            }}
          />
          <span className="ml-2">{label}</span>
        </label>
      ))}
```

... and somewhere before define

```ts
const options = [
  { label: 'Slide from left', value: 'left' },
  { label: 'Slide from right', value: 'right' },
]
```

File: apps/preview/next/pages/showcases/Drawer/TransitionAndCloseButton.tsx [hunk 2]
```diff
@@ -1,70 +1,105 @@
...
+  SfDrawerProps,
+} from '@storefront-ui/react';
 import { useState, useRef } from 'react';
-import { CSSTransition } from 'react-transition-group';
+import { Transition } from 'react-transition-group';
+import classNames from 'classnames';
 
 export default function DrawerWithTransition() {
   const [open, setOpen] = useState(false);
+  const [checkedPlacement, setCheckedPlacement] = useState<SfDrawerProps['placement']>('left');
```
Comment (dkacper) [inline apps/preview/next/pages/showcases/Drawer/TransitionAndCloseButton.tsx:19]: Could you rename it to just `placement`? I think it'd be easier to scan the code later to find which parts relates to 'left' and which to 'right' variant. Also, there's an enum `SfDrawerPlacement` that you can use to define the type here.

```suggestion
  const [placement, setPlacement] = useState<`${SfDrawerPlacement}`>('left');
```

File: apps/preview/nuxt/pages/showcases/Drawer/TransitionAndCloseButton.vue [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+    <label class="flex items-center my-4 cursor-pointer"
+      ><SfRadio v-model="placement" class="flex items-center" name="placement" value="left" /><span class="ml-2"
+        >Slide from left</span
+      ></label
+    >
+    <label class="flex items-center my-4 cursor-pointer"
+      ><SfRadio v-model="placement" class="flex items-center" name="placement" value="right" /><span class="ml-2"
+        >Slide from right</span
+      ></label
+    >
```
Comment (dkacper) [inline apps/preview/nuxt/pages/showcases/Drawer/TransitionAndCloseButton.vue:12]: You could also use `v-for` here.

**Review by Szymon-dziewonski — APPROVED** [review]
From my side this looks good, @FRSgit can you please confirm if `scale(1)` could be removed?

File: apps/preview/nuxt/pages/showcases/Drawer/TransitionAndCloseButton.vue [hunk 4]
```diff
@@ -1,13 +1,20 @@
 <template>
+  <fieldset>
+    <label v-for="{ label, value } in options" :key="value" class="flex items-center my-4 cursor-pointer"
+      ><SfRadio v-model="placement" class="flex items-center" name="placement" :value="value" /><span class="ml-2">{{
+        label
+      }}</span></label
+    >
```
Comment (FRSgit) [inline apps/preview/nuxt/pages/showcases/Drawer/TransitionAndCloseButton.vue:7]: The autoformat is pretty weird here, can we rewrite it like:
```suggestion
    <label v-for="{ label, value } in options" :key="value" class="flex items-center my-4 cursor-pointer">
      <SfRadio v-model="placement" class="flex items-center" name="placement" :value="value" />
      <span class="ml-2">{{ label }}</span>
    </label>
```

**Review by dkacper — APPROVED** [review]
LGTM ⭐

**Review by Szymon-dziewonski — APPROVED** [review]

**Pull request merged**